app/Services/LoanService.php

```diff
@@ -32,10 +75,64 @@ public function createLoan(User $user, int $amount, string $currencyCode, int $t
      * @param  string  $currencyCode
      * @param  string  $receivedAt
      *
-     * @return ReceivedRepayment
+     * @return ReceivedRepayment|Model
      */
     public function repayLoan(Loan $loan, int $amount, string $currencyCode, string $receivedAt): ReceivedRepayment
     {
-        //
+        $receivedRepayment = $loan->receivedRepayments()->create([
+            'amount' => $amount,
+            'currency_code' => $currencyCode,
+            'received_at' => Carbon::createFromFormat('Y-m-d', $receivedAt)->format('Y-m-d'),
+        ]);
+        if ($receivedRepayment) {
+            Log::info('Store received repayment success!');
+            Log::debug('Received repayment data: ' . json_encode($receivedRepayment));
```

Comment:
👍  for logging
probably could use some love to improve the structure for later consuption